src/FsAutoComplete/CodeFixes/ConvertTripleSlashCommentToXmlTaggedDoc.fs

```diff
@@ -0,0 +1,217 @@
+module FsAutoComplete.CodeFix.ConvertTripleSlashCommentToXmlTaggedDoc
+
+open FsToolkit.ErrorHandling
+open FsAutoComplete.CodeFix.Types
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text.Range
+open FSharp.Compiler.Xml
+open System
+
+let title = "Convert '///' comment to XML-tagged doc comment"
+
+let private containsPosAndNotEmptyAndNotElaborated (pos: FSharp.Compiler.Text.Position) (xmlDoc: PreXmlDoc) =
+  let d = xmlDoc.ToXmlDoc(false, None)
+
+  if rangeContainsPos d.Range pos then
+    let summaryPresent =
+      d.UnprocessedLines |> Array.exists (fun s -> s.Contains("<summary>"))
+
+    not xmlDoc.IsEmpty && not summaryPresent
+  else
+    false
+
+let private isLowerAstElemWithPreXmlDoc input pos =
+  SyntaxTraversal.Traverse(
+    pos,
+    input,
+    { new SyntaxVisitorBase<_>() with
+        member _.VisitBinding(_, defaultTraverse, synBinding) =
+          match synBinding with
+          | SynBinding(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+          | _ -> defaultTraverse synBinding
+
+        member _.VisitComponentInfo(_, synComponentInfo) =
+          match synComponentInfo with
+          | SynComponentInfo(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
+          | _ -> None
+
+        member _.VisitInheritSynMemberDefn(_, _, _, _, members, _) =
+          let isInLine c =
+            match c with
+            // ToDo not working
+            | SynMemberDefn.AutoProperty(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc ->
+              Some xmlDoc
```

Comment:
do you think this is a bug in the FCS Walker that makes auto-properties not get traversed?

Reply:
The whole VisitInheritSynMemberDefn member seems not to be needed, so I removed it.

I added the following test to `TreeVisitorTests.fs` in the fsharp repo and it fails. But I have really no idea if this API is even meant to be used this way.

```fsharp
[<Test>]
let ``Visit AutoProperty`` () =
    let pos = mkPos 2 13
    let containsPosAndNotEmptyAndNotElaborated (pos: FSharp.Compiler.Text.Position) (xmlDoc: PreXmlDoc) =
      let containsPosAndNoSummaryPresent (xd: PreXmlDoc) =
        let d = xd.ToXmlDoc(false, None)

        if SyntaxTraversal.rangeContainsPosEdgesExclusive d.Range pos then
          let summaryPresent =
            d.UnprocessedLines |> Array.exists (fun s -> s.Contains("<summary>"))

          not summaryPresent
        else
          false

      not xmlDoc.IsEmpty && containsPosAndNoSummaryPresent xmlDoc
  
    let visitor =
        { new SyntaxVisitorBase<_>() with
            member _.VisitInheritSynMemberDefn(_, _, _, _, members, _) =
                let isInLine c =
                    match c with
                    | SynMemberDefn.AutoProperty(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
                    | SynMemberDefn.ImplicitCtor(xmlDoc = xmlDoc) when containsPosAndNotEmptyAndNotElaborated pos xmlDoc -> Some xmlDoc
                    | _ -> None
                members |> List.tryPick isInLine
                     }

    let source = """
type MyClass() =
    /// fails on AutoProperty
    member val Name = "" with get, set
"""
    let parseTree = parseSourceCode("C:\\test.fs", source)

    match SyntaxTraversal.Traverse(pos, parseTree, visitor) with
    | Some xml -> ()
    | _ -> failwith "Did not visit AutoProperty"

```